### PR TITLE
First implementation of custom keyboard mapping

### PR DESCRIPTION
**Pull request by k0pernicus**
* You can remap the keys on your keyboard in the new package kb_layouts, in ```drivers/```
* A French layout has been implemented (so cool to me!)
* TODO : To change the layout from GUI/Terminal

### CODE REVIEW

**Comment by Ticki** [comment]
This is damn cool

**Comment by k0pernicus** [comment]
Thanks!
I must think about changing the keyboard layout in the current session now... :-)

**Comment by jackpot51** [comment]
I have an issue with shift after using this

File: kernel/drivers/kb_layouts/layouts.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+/// Function to return the character associated with the scancode, and the layout
+pub fn char_for_scancode(scancode: u8, shift: bool, layout: &Layout) -> char {
+    let mut character = '\x00';
+    if scancode < 58 {
+        let characters: [char; 2] = get_scancode_from_layout(layout, scancode);
+        if shift {
+            character = characters[1]
+        }
+        //Else...
```
Comment (jackpot51) [inline kernel/drivers/kb_layouts/layouts.rs]: This should be an else

**Comment by k0pernicus** [comment]
Crap! Sorry.

**Comment by k0pernicus** [comment]
I fixed it, I push...

File: kernel/drivers/ps2.rs [hunk 2]
```diff
@@ -46,6 +51,7 @@ impl Ps2 {
             mouse_i: 0,
             mouse_x: 0,
             mouse_y: 0,
+            layout: layouts::Layout::FRENCH,
```
Comment (jackpot51) [inline kernel/drivers/ps2.rs]: The default is still set to french. Perhaps a #[cfg(language)] could be used somehow?

**Comment by k0pernicus** [comment]
Good idea - I will work on it tomorrow ;-)

**Pull request merged**